src/neuroconv/tools/nwb_helpers/_metadata_and_file_helpers.py

```diff
@@ -221,27 +222,35 @@ def make_or_load_nwbfile(
 
         io = backend_io_class(**load_kwargs)
 
+    nwbfile_loaded_succesfully = True
     try:
         if load_kwargs.get("mode", "") == "r+":
             nwbfile = io.read()
         elif nwbfile is None:
             nwbfile = make_nwbfile_from_metadata(metadata=metadata)
         yield nwbfile
-    except Exception as e:
-        success = False
-        raise e
+    except Exception as load_error:
+        nwbfile_loaded_succesfully = False
+        raise load_error
     finally:
         if nwbfile_path_in:
+            nwbfile_written_succesfully = True
             try:
-                if success:
+                if nwbfile_loaded_succesfully:
                     io.write(nwbfile)
 
                     if verbose:
                         print(f"NWB file saved at {nwbfile_path_in}!")
+            except Exception as write_error:
+                nwbfile_written_succesfully = False
+                raise write_error
             finally:
                 io.close()
 
-                if not success and not file_initially_exists:
+                if not nwbfile_loaded_succesfully and not file_initially_exists:  # Not sure we need this
```

Comment:
If no file initially exists then there is nothing to unlink?

Reply:
Makes sense.